benchmarks/time/src/main/scala/strawman/collection/immutable/ImmutableArrayBenchmark.scala

```diff
@@ -74,19 +96,40 @@ class ImmutableArrayBenchmark {
   }
 
   @Benchmark
-  def prependAll(bh: Blackhole): Unit = bh.consume(xs ++: xs)
+  @OperationsPerInvocation(1000)
+  def prependAll(bh: Blackhole): Unit = {
+    var ys = fresh(size)
+    val zs = fresh((size / 1000) max 1)
```

Comment:
Besides the problem I commented on above concerning sharing of the `xs` field, the problem with the `xss` array was that it took a really long time to create for the larger benchmark sizes. Perhaps a single `zs` field could be used (but this might also be effected by the sharing problem).